packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -821,4 +827,51 @@ export function deriveEstimateSizeOptions(scope: IConstruct): EstimateSizeOption
   }
 
   return { actionEstimate, arnEstimate };
+}
+
+/**
+ * A class that behaves both as a set and an array
+ *
+ * Used for the elements of a PolicyStatement. In practice they behave as sets,
+ * but we have thousands of snapshot tests in existence that will rely on a
+ * particular order so we can't just change the type to `Set<>` wholesale without
+ * causing a lot of churn.
+ */
+class OrderedSet<A> {
+  private readonly set = new Set<A>();
+  private readonly array = new Array<A>();
+
+  /**
+   * Add new elements to the set
+   *
+   * Returns the elements actually added
+   */
+  public push(...xs: A[]) {
```

Comment:
```suggestion
  /**
   * Add new elements to the set
   *
   * @param xs the elements to be added
   *
   * @returns the elements actually added
   */
  public push(...xs: readonly A[]): A[] {
```

As far as the "set" semantics are concerned, this should probably be named `add` instead of `push`...